saturday-afternoon/components/flights/FlightItem.js

```diff
@@ -21,32 +21,42 @@ const buttonStyles = resolveScopedStyles(
   </scope>
 );
 
-const FlightItem = ({ flight }: Props) => (
-  <div>
-    <Row>
-      <Legs legs={flight.legs} />
-    </Row>
-    <Row>
-      <Button type="primary" className={`buy ${buttonStyles.className}`}>
-        Buy for {flight.price.amount} {flight.price.currency}
-      </Button>
-    </Row>
-    {buttonStyles.styles}
-  </div>
-);
+const FlightItem = ({ flight }: Props) => {
+  const onClick = () => flight.bookingUrl && window.open(flight.bookingUrl);
```

Comment:
ahhhmm, do you think it is a good idea to use a callback in render like this? and the function name should be different (handleBuy?)

Reply:
no, it's not ideal, unless I transform it into a class component, do you have a better idea? :)